src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -57,9 +57,9 @@ pub struct CloudwatchLogsSinkConfig {
     pub encoding: Encoding,
     pub create_missing_group: Option<bool>,
     pub create_missing_stream: Option<bool>,
-    #[serde(default, flatten)]
+    #[serde(default)]
     pub batch: BatchConfig,
-    #[serde(flatten)]
+    #[serde(default)]
```

Comment:
Why do we need to start defaulting this (and all the others)?

Reply:
I think `serde` handles options here differently than types that implement `Deserialize + Default`. In that case you need to tell it to do default I believe. In my testing adding this gave us what we wanted while not having it didn't.

Reply:
My concern is that this will always give us the `Default` impl for `TowerRequestConfig` instead of something empty that we can fill with the per-sink default. Unless the default is an empty value?

Reply:
We don't use `Default` for sink specific defaults, we use `TowerRequestConfig::unwrap_with` to set sink specific defaults. So I think this isn't an issue.

Reply:
Without the `serde(default)`, if none of the values in `request` are configured, serde errors with a "missing field" error on `request`. See [this Rust playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=68c4c9153243ae4457cff628e2e78bc3)

Reply:
I see, the default is all fields set to `None` and we merge in the per-sink defaults.